core/editor/src/tools/mod.rs

```diff
@@ -1,14 +1,32 @@
-use crate::events::{ModKeys, MouseState, TracePoint};
-use crate::{events::Trace, Color};
+mod crop;
+mod ellipse;
+mod line;
+mod navigate;
+mod path;
+mod pen;
+mod rectangle;
+mod sample;
+mod select;
+mod shape;
+
+use crate::events::{Event, ModKeys, MouseState, Trace, TracePoint};
+use crate::Color;
+use crate::EditorError;
+use document_core::Operation;
 use std::collections::HashMap;
 
+pub trait Tool {
+	fn handle_input(&mut self, event: Event) -> Option<Operation>;
+}
+
 pub struct ToolState {
 	pub mouse_state: MouseState,
 	pub mod_keys: ModKeys,
 	pub trace: Trace,
 	pub primary_color: Color,
 	pub secondary_color: Color,
-	pub active_tool: ToolType,
+	pub active_tool_type: ToolType,
+	pub tools: HashMap<ToolType, Box<dyn Tool>>,
```

Comment:
Does it make sense to store this on `ToolState`? It doesn't seem like state to me.

Reply:
Yes it does, pleas @ me for details